telemetry/definitions/commonDefinitions.json

```diff
@@ -1319,6 +1324,32 @@
             "description": "Open a window to view/edit IAM User Configuration",
             "metadata": [{ "type": "result" }]
         },
+        {
+            "name": "iam_open",
+            "description": "Open a window to view/edit an IAM resource",
+            "metadata": [
+                { "type": "result" },
+                { "type": "iamResourceType" }
+            ]
+        },
+        {
+            "name": "iam_create",
+            "description": "Create an IAM resource",
+            "metadata": [
+                { "type": "result" },
+                { "type": "iamResourceType" },
+                { "type": "reason", "required": false }
```

Comment:
Reason can now be left out of the definition -- it is considered implicit (and the C# generator adds it to the base metric class as an optional field)